src/ethereum_fmt_plug.c

```diff
@@ -245,6 +246,7 @@ struct fmt_main fmt_ethereum = {
 		ethereum_get_binary,
 		ethereum_common_get_salt,
 		{
+			ethereum_common_kdf_type,
 			ethereum_common_iteration_count,
```

Comment:
What's the difference between kdf 0 and 2? Both are "PBKDF2-SHA256" or is that a typo?

Also, the cost with most impact should be listed first. I'm not sure but perhaps iteration count should be first.

Reply:
Yes, both are "PBKDF2-SHA256" and it is not a typo. One of them has SIMD support, and the other doesn't (due to varying salt usage). Perhaps I can distinguish between them by injecting the word "SIMD" in the KDF description string?

I can change the commit to mention the `iteration count` cost first. This idea makes sense to me.

Reply:
> One of them has SIMD support, and the other doesn't (due to varying salt usage). Perhaps I can distinguish between them by injecting the word "SIMD" in the KDF description string?

No, but perhaps some other descriptive comment. Not quite sure what. Are these different versions in some sense that would make sense to an end user?

Reply:
The usage of "SIMD" string might not make much sense to the user in a direct manner, but it may help in determining why there is a speed difference between `type 0` and `type 2` hashes if the user is a bit curious. I am OK with dropping this idea :-)

Reply:
Perhaps rename this cost to "type" instead of KDF. The "2" would then be "presale" and so on.

Reply:
Yes, mapping "2" to "presale" is relatively straightforward. However, what I should call the "0" and "1" entries? I don't have good answers right now.

I think exposing the KDF makes more sense from a cost analysis point-of-view.

Reply:
Maybe keep it as KDF and have 2 as "PBKDF2-SHA256 presale".

Reply:
Sounds good. Doing so now.